src/pages/registration/contributors_tab/components/SelectAffiliationsCheckbox.tsx

```diff
@@ -0,0 +1,55 @@
+import CheckBoxIcon from '@mui/icons-material/CheckBox';
+import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank';
+import { IconButton } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { CristinPerson, CristinPersonAffiliation } from '../../../../types/user.types';
+import { dataTestId } from '../../../../utils/dataTestIds';
+
+interface SelectAffiliationsCheckboxProps {
+  personIsSelected: boolean;
+  affiliation: CristinPersonAffiliation;
+  selectedPerson?: CristinPerson;
+  setSelectedPerson: (selectedContributor: CristinPerson | undefined) => void;
+  affiliationIsSelected: boolean;
+}
+
+export const SelectAffiliationsCheckbox = ({
+  affiliation,
+  selectedPerson,
+  setSelectedPerson,
+  affiliationIsSelected,
+  personIsSelected,
+}: SelectAffiliationsCheckboxProps) => {
+  const { t } = useTranslation();
+
+  const selectPerson = () => {
+    if (!selectedPerson) {
+      return;
+    }
+    const newAffiliations = affiliationIsSelected
+      ? selectedPerson.affiliations.filter((a) => a.organization !== affiliation.organization)
+      : [...selectedPerson.affiliations, affiliation];
+
+    const personWithAffiliation: CristinPerson = {
+      ...selectedPerson,
+      affiliations: newAffiliations,
+    };
+    setSelectedPerson(personWithAffiliation);
+  };
+
+  return (
+    <IconButton
```

Comment:
Vanlig å bruke `<Checkbox />` direkte i stedet for å selv håndtere check osv med `Button` og ikon, men mulig det er noe custom håndtering vi ønsker?

Reply:
Ja jeg vet ikke hva som var tanken i den opprinnelige koden, dette er kode som jeg helst rører minst mulig fordi det er en komponent jeg "låner" inn i min kode